### ISSUE TITLE
Request for Feedback: V1.0.7 release

### ISSUE THREAD

**Issue by guwirth** [etas/vs-boost-unit-test-adapter#126]
Hi,

After some final testing I would release the current version 1.0.7. Think this should be the last version supporting old Visual Studio and Boost versions.

With next 1.0.8 I would then:
* support VS2015+ only / remove support of older VS versions
* support Boost 1.60+  only / remove support of older Boost versions
* remove parsing the source code

Regards,
Günter

**Comment by guwirth** [comment]
Moved remaining issues to v1.0.8.

**Comment by guwirth** [comment]
Release Notes: https://github.com/etas/vs-boost-unit-test-adapter/releases/tag/buta-1.0.7

**Comment by guwirth** [comment]
Hi everyone,

Like to ask you to test v1.0.7. If everything is working we will distribute it in the Visual Studio Gallery:
https://github.com/etas/vs-boost-unit-test-adapter/releases/tag/buta-1.0.7

Regards,

**Comment by foobar303** [comment]
Hi,

tested today Buta 1.0.7. using Microsoft Visual Studio Ultimate 2013, Version 12.0.40629.00 Update 5 and my existing tests. Unfortunatly I cannot see let alone execute my data driven tests via the test adapter. As by your discussion on this topic, issue #23 _Support BOOST_DATA_TEST_CASE_ should have been solved in 1.0.7. Any feedback on how to use this feature is greatly appreciated, since this (beside supporting DISABLED tests) is the one I was looking forward for a rather long time. 

However, at least I detected no breaking changes with my tests so far :)

Regards

**Comment by guwirth** [comment]
@foobar303 thanks for your feedback and just for completeness: which boost version are you using?

**Comment by foobar303** [comment]
Boost 1.60.0.

**Comment by foobar303** [comment]
Additional note: Test output yields the following message "_Could not locate debug symbols for 'C:\workspaces.build\bin\Debug\MyApp.exe'. To make use of '--list_content' discovery, ensure that debug symbols are available or make use of '<ForceListContent>' via a .runsettings file._"

**Comment by netspiri** [comment]
Hi @foobar303 and thank you for the feedback.

Could you kindly look into issue #120 and see if your issue is similar? In brief, list content discovery is not being used due to missing debug symbols which was caused by certain compilation properties.

**Comment by foobar303** [comment]
Which compiler options are known to cause failing the list content discovery? My test projects use the following settings:

- Linker/Debugging/Generate Debug Info: "Yes (/DEBUG)" 
- C/C++/General/Debug Information Format: "Program Database (/Zi)". I tried /Z7 and  /ZI instead (while the latter will be changed back automatically since I'm compiling for 64 bit), anyway, none of these changed the described behavior.

Eventually, when using a .runsettings file
```
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <BoostTest>
    <ForceListContent>true</ForceListContent>
  </BoostTest>
</RunSettings>
```
the warning mentioned in my previous post disappeared. Although in this case my data driven tests do not show up in Test Explorer either (am I wrong to expect this?), the VS Test Output showed messages like the following when running all tests from Test Explorer:
```
Executing:   -> [UnitTests_DbUtils/ScanMode_WithinBounds_DoesNotThrow]
Executing:   -> [UnitTests_DbUtils/ScanMode_OutOfBounds_Throws]
Test adapter sent back a result for an unknown test case. Ignoring result for 'ScanMode_WithinBounds_DoesNotThrow'.
Test adapter sent back a result for an unknown test case. Ignoring result for 'ScanMode_OutOfBounds_Throws'.
```
Guess this indicates that at least test have been being executed but for some reason test adapter can't handle the results properly. Note that test do not fail when executing via the .exe.

**Comment by netspiri** [comment]
@foobar303 Could you kindly open a new issue(s) in this regard please? It seems that there are 2 issues: debug symbols not being located properly and possibly something wrong with test registration for BOOST_DATA_TEST_CASE.

Providing log files (log file path should be listed in output window) should be helpful for debugging. A test sample which can be used to reproduce the issue also helps. Please note that these may contain sensitive data (path information, test names, source code) so be wary before posting such material.

**Comment by foobar303** [comment]
Hi netspiri,

just opened new issu #130.

**Comment by guwirth** [comment]
Team 1: Visual Studio 2015, Boost 1.60 ==> everything working

**Comment by guwirth** [comment]
Team 2: Visual Studio 2015, Boost 1.59 ==> Breaking change #71, #99

With 1.0.6 the "current directory" was the directory of the boost EXE. Starting with 1.0.7 it's the `Config / Debugging / Working Directory` setting which is per default `$(ProjectDir)` and not `$(OutDir)` as before.

Don't know where VS is storing this setting? Seems to be not in the project file .vcxproj

**Comment by mwu-tow** [comment]
@guwirth 
VS by default stores the Working Directory setting in the `ProjectName.vcxproj.user` file that is placed next to the project file. The relevant XML node is named `LocalDebuggerWorkingDirectory`.

Answering the main question — 1.0.7 changes that detect working directory and environment work great for me. Thanks! :-)

**Comment by netspiri** [comment]
@foobar303 Unfortunately, that is a known issue. The Boost test framework (1.59) returns that kind of data as the line number (see example output below). I do not know if this has been fixed in recent Boost Test versions. Regarding the use of .runsettings, it can be avoided if you include the test framework within your module instead of dynamically linking to Boost Test. By omitting the `#define BOOST_TEST_DYN_LINK` and providing debug symbols, the test adapter should be able to use `--list_content=DOT` automatically.

```DOT
tu65581[shape=Mrecord,fontname=Helvetica,color=yellow,label="DataLost|d:\boostunittest\test_a.cpp(1047)"];
tu22 -> tu65581;
tu65590[shape=Mrecord,fontname=Helvetica,color=yellow,label="ChangeEvent|(3435973836)"];
tu22 -> tu65590;
```

@guwirth As @mwu-tow correctly mentioned, the property can also be set from here:
![projectproperties](https://cloud.githubusercontent.com/assets/6962427/16142165/4f35835a-3463-11e6-993b-547290edc6bb.png)
The reason for this 'breaking' change in behaviour is to ensure that running the test program from Visual Studio (via Debug - Start Debugging/Start Without Debugging) and from the test explorer should react in an equivalent manner. We have had issues in the past were tests running via the test explorer passed but running them from the executable directly from Visual Studio failed due to assumptions made about relative paths.

As a general note, I would like to thank all for the helpful feedback and for supporting this project.

**Comment by guwirth** [comment]
> ... The relevant XML node is named LocalDebuggerWorkingDirectory. ...

**Comment by guwirth** [comment]
> @guwirth As @mwu-tow correctly mentioned, the property can also be set from here:

Think current behaviour is ok but surprising for some users because their unit test are red now. Setting can be changed per project but only in .user file which is currently not part of our repository.

Main issue is that behaviour from command line and within VS is different now.:
* run tests from command line: test files are expected in parallel to EXE
* run test from VS: test files are expected in $(ProjectDir) per default

... but at the end I think we should release it as it is.

**Comment by guwirth** [comment]
**VERSION 1.0.7.57** is ready to download from the Visual Studio Gallery.
Also from my side thanks to all contributors!
